kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -269,63 +269,76 @@ ip4configProperties, new AccessPointsProperties(activeAccessPoint, accessPoints)
     }
 
     public synchronized void apply(Map<String, Object> networkConfiguration) throws DBusException {
-        doApply(networkConfiguration);
-        this.cachedConfiguration = networkConfiguration;
+        try {
+            configurationEnforcementDisable();
+            doApply(networkConfiguration);
+            this.cachedConfiguration = networkConfiguration;
+        } finally {
+            configurationEnforcementEnable();
+        }
     }
 
     public synchronized void apply() throws DBusException {
         if (Objects.isNull(this.cachedConfiguration)) {
             logger.warn("No cached network configuration found.");
             return;
         }
-        doApply(this.cachedConfiguration);
-    }
-
-    private synchronized void doApply(Map<String, Object> networkConfiguration) throws DBusException {
-        logger.info("Applying configuration using NetworkManager Dbus connector");
         try {
             configurationEnforcementDisable();
             modemResetHandlersDisable();
-
-            NetworkProperties properties = new NetworkProperties(networkConfiguration);
-
-            List<String> configuredInterfaces = properties.getStringList("net.interfaces");
-            manageConfiguredInterfaces(configuredInterfaces, properties);
-
-            List<Device> availableInterfaces = getAllDevices();
-            manageNonConfiguredInterfaces(configuredInterfaces, availableInterfaces);
+            doApply(this.cachedConfiguration);
         } finally {
             configurationEnforcementEnable();
         }
     }
 
-    private synchronized void manageConfiguredInterfaces(List<String> configuredInterfaces,
-            NetworkProperties properties) throws DBusException {
-        List<String> availableInterfaces = getDeviceIds();
-
-        for (String iface : configuredInterfaces) {
-            if (!availableInterfaces.contains(iface)) {
-                logger.debug("Configured device \"{}\" not available on the system. Ignoring configuration.", iface);
-                continue;
-            }
+    public synchronized void apply(String deviceId) throws DBusException {
+        if (Objects.isNull(deviceId) || deviceId.isEmpty()) {
+            throw new IllegalArgumentException("DeviceId cannot be null or empty.");
+        }
+        if (Objects.isNull(this.cachedConfiguration)) {
+            logger.warn("No cached network configuration found.");
+            return;
+        }
+        try {
+            configurationEnforcementDisable();
+            doApply(deviceId, this.cachedConfiguration);
```

Comment:
You need to clear possible pending reset timers here:

```suggestion
            configurationEnforcementDisable();
            modemResetHandlersDisable(deviceId);
            doApply(deviceId, this.cachedConfiguration);
```

Currently pending timers are tracked in a `List<NMModemResetHandler>`. I think you just need to move to a `Map<String, NMModemResetHandler>` so that you can keep track of the available reset handlers and remove only the ones you need